pkg/controller/cloneset/sync/cloneset_update.go

```diff
@@ -311,37 +230,66 @@ func (c *realControl) updatePod(cs *appsv1alpha1.CloneSet, coreControl clonesetc
 
 	klog.V(2).Infof("CloneSet %s/%s start to patch Pod %s specified-delete for update %s", cs.Namespace, cs.Name, pod.Name, updateRevision.Name)
 
-	if patched, err := specifieddelete.PatchPodSpecifiedDelete(c, pod, "true"); err != nil {
+	if patched, err := specifieddelete.PatchPodSpecifiedDelete(c.Client, pod, "true"); err != nil {
 		c.recorder.Eventf(cs, v1.EventTypeWarning, "FailedUpdatePodReCreate",
 			"failed to patch pod specified-delete %s for update(revision %s): %v", pod.Name, updateRevision.Name, err)
 		return 0, err
 	} else if patched {
 		clonesetutils.ResourceVersionExpectations.Expect(pod)
 	}
 
-	//clonesetutils.ScaleExpectations.ExpectScale(clonesetutils.GetControllerKey(cs), expectations.Delete, pod.Name)
-	//if err := c.Delete(context.TODO(), pod); err != nil {
-	//	clonesetutils.ScaleExpectations.ObserveScale(clonesetutils.GetControllerKey(cs), expectations.Delete, pod.Name)
-	//	c.recorder.Eventf(cs, v1.EventTypeWarning, "FailedUpdatePodReCreate",
-	//		"failed to delete pod %s for update: %v", pod.Name, err)
-	//	return 0, err
-	//}
-	//
-	//// TODO(FillZpp): add a strategy controlling if the PVCs of this pod should be deleted
-	//for _, pvc := range pvcs {
-	//	if pvc.Labels[appsv1alpha1.CloneSetInstanceID] != pod.Labels[appsv1alpha1.CloneSetInstanceID] {
-	//		continue
-	//	}
-	//
-	//	clonesetutils.ScaleExpectations.ExpectScale(clonesetutils.GetControllerKey(cs), expectations.Delete, pvc.Name)
-	//	if err := c.Delete(context.TODO(), pvc); err != nil {
-	//		clonesetutils.ScaleExpectations.ObserveScale(clonesetutils.GetControllerKey(cs), expectations.Delete, pvc.Name)
-	//		c.recorder.Eventf(cs, v1.EventTypeWarning, "FailedDelete", "failed to delete pvc %s: %v", pvc.Name, err)
-	//		return 0, err
-	//	}
-	//}
-
 	c.recorder.Eventf(cs, v1.EventTypeNormal, "SuccessfulUpdatePodReCreate",
 		"successfully patch pod %s specified-delete for update(revision %s)", pod.Name, updateRevision.Name)
 	return 0, nil
 }
+
+func sortUpdateIndexes(coreControl clonesetcore.Control, strategy appsv1alpha1.CloneSetUpdateStrategy, pods []*v1.Pod, waitUpdateIndexes []int) []int {
+	// Sort Pods with default sequence
+	sort.Slice(waitUpdateIndexes, coreControl.GetPodsSortFunc(pods, waitUpdateIndexes))
+
+	if strategy.PriorityStrategy != nil {
+		waitUpdateIndexes = updatesort.NewPrioritySorter(strategy.PriorityStrategy).Sort(pods, waitUpdateIndexes)
+	}
+	if strategy.ScatterStrategy != nil {
+		waitUpdateIndexes = updatesort.NewScatterSorter(strategy.ScatterStrategy).Sort(pods, waitUpdateIndexes)
+	}
+
+	// PreparingUpdate first
+	sort.Slice(waitUpdateIndexes, func(i, j int) bool {
```

Comment:
what is the relative weights of different sort criteria: PriorityStrategy , ScatterStrategy and GetPodLifecycleState?
Is it better to use stable sort (slice -> sliceStable) here ?
please describe them in the func comments.